api/app/controllers/spree/api/v2/storefront/cart_controller.rb

```diff
@@ -216,6 +217,13 @@ def load_variant
             @variant = current_store.variants.find(params[:variant_id])
           end
 
+          def ensure_valid_metadata
```

Comment:
@aleksandarpetrushev 

1. Let's move that into a concern `Spree::Api::V2::Storefront::Metadata`
2. Please use translations
3. Please use https://github.com/spree/spree/blob/9d335d35583ae0783fda542d46053f642c3b8dfe/api/app/controllers/spree/api/v2/base_controller.rb#L55